apps/ledger-live-mobile/src/logic/terms.tsx

```diff
@@ -58,10 +64,19 @@ async function unAcceptTerms() {
   await AsyncStorage.removeItem("acceptedTermsVersion");
 }
 
-export async function acceptTerms() {
+async function acceptTerms() {
   await AsyncStorage.setItem("acceptedTermsVersion", currentTermsRequired);
 }
 
+const acceptTermsAndUpdateStateRef: MutableRefObject<(() => void) | null> =
+  React.createRef();
+
+export function acceptTermsAndUpdateState() {
+  if (acceptTermsAndUpdateStateRef.current)
+    acceptTermsAndUpdateStateRef.current();
+  else acceptTerms();
```

Comment:
Same here, a small comment to explain why this is necessary ? If the ref is not yet set (it should be set in the useEffect) we don't set the state `accepted` ? But why ?